tests/api/v2/handlers/test_operations_api.py

```diff
@@ -0,0 +1,241 @@
+                               cleanup=0, pin=0, host=test_agent.host, deadman=False, used=[], id='456',
+                               relationships=[]))
+    test_link.host = test_agent.host
+    test_operation.chain.append(test_link)
+    loop.run_until_complete(BaseService.get_service('data_svc').store(test_operation))
+
+
+@pytest.mark.usefixtures(
+    "setup_operations_api_test"
+)
+class TestOperationsApi:
+    async def test_get_operations(self, api_client, api_cookies):
+        resp = await api_client.get('/api/v2/operations', cookies=api_cookies)
+        operations_list = await resp.json()
+        assert len(operations_list) == 1
+        operation_dict = operations_list[0]
+        assert operation_dict['name'] == 'My Test Operation'
+        assert operation_dict['id'] == '123'
+
+    async def test_unauthorized_get_operations(self, api_client):
+        resp = await api_client.get('/api/v2/operations')
+        assert resp.status == HTTPStatus.UNAUTHORIZED
+
+    async def test_get_operation_by_id(self, api_client, api_cookies):
+        resp = await api_client.get('/api/v2/operations/123', cookies=api_cookies)
+        operation_dict = await resp.json()
+        assert operation_dict['name'] == 'My Test Operation'
+        assert operation_dict['id'] == '123'
+
+    async def test_unauthorized_get_operation_by_id(self, api_client):
+        resp = await api_client.get('/api/v2/operations/123')
+        assert resp.status == HTTPStatus.UNAUTHORIZED
+
+    async def test_get_operation_report(self, api_client, api_cookies, mocker, async_return):
+        with mocker.patch('app.objects.c_operation.Operation.all_facts') as mock_all_facts:
+            mock_all_facts.return_value = async_return([])
+            test_objective = Objective(id='123', name='test objective', description='test', goals=[])
+            BaseService.get_service('data_svc').ram['operations'][0].objective = test_objective
+            resp = await api_client.get('/api/v2/operations/123/report', cookies=api_cookies)
+            report = await resp.json()
+            assert report['name'] == 'My Test Operation'
+            assert report['jitter'] == '2/8'
+
+    async def test_unauthorized_get_operation_report(self, api_client):
+        resp = await api_client.get('/api/v2/operations/report')
+        assert resp.status == HTTPStatus.UNAUTHORIZED
+
+    async def test_create_operation(self, api_client, api_cookies):
+        payload = dict(name='post_test', planner={'id': '123'},
+                       adversary={'adversary_id': '123'}, source={'id': '123'})
+        resp = await api_client.post('/api/v2/operations', cookies=api_cookies, json=payload)
+        assert resp.status == HTTPStatus.OK
+        op_data = await resp.json()
+        assert op_data.get('name') == "post_test"
+        op_exists = await BaseService.get_service('data_svc').locate('operations', {'name': 'post_test'})
+        assert op_exists
+
```

Comment:
we can also test trying to create an operation using an ID that's already in use